tgui/packages/tgui/interfaces/MutationSelector.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  Button,
+  Collapsible,
+  Flex,
+  ProgressBar,
+  Section,
+  Tooltip,
+} from 'tgui-core/components';
+
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+
+type Upgrade = {
+  name: string;
+  desc: string;
+  owned: boolean;
+};
+
+type BiomassData = {
```

Comment:
btw for these, i scanned and it looked fine but remember that you need to do var?: type; for when it can be possibly not sent/null

Reply:
in that case this is all correct